src/autocompleteDump.ts

```diff
@@ -0,0 +1,207 @@
+import * as request from "request-promise-native"
+import * as vscode from "vscode"
+import { parseStringPromise } from "xml2js"
+
+const AUTOCOMPLETE_METADATA = "https://raw.githubusercontent.com/CloneTrooper1019/Roblox-Client-Tracker/roblox/AutocompleteMetadata.xml"
+
+export interface AutocompleteParameter {
+    type: string,
+    name: string,
+    optional: boolean,
+    constraint?: string,
+}
+
+export interface AutocompleteReturn {
+    type: string,
+    name?: string,
+}
+
+export interface AutocompleteFunction {
+    name: string,
+    static: boolean,
+    description?: string,
+    parameters: AutocompleteParameter[],
+    returns: AutocompleteReturn[],
+}
+
+export interface AutocompleteProperty {
+    name: string,
+    static: boolean,
+    type: string,
+    description?: string,
+}
+
+export interface AutocompleteGroup {
+    name: string,
+    functions: AutocompleteFunction[],
+    properties: AutocompleteProperty[],
+}
+
+export interface AutocompleteDump {
+    LuaLibrary: AutocompleteGroup[],
+    ItemStruct: AutocompleteGroup[],
+}
+
+// Interfaces for the data provided in the XML parser
+interface XMLPropertyPart {
+    _: string, // Description
+    $: {
+        name: string,
+        static?: string,
+    }
+}
+
+interface XMLOrderedPropertyPart extends XMLPropertyPart {
+    "#name": string // Property type
+}
+
+interface XMLProperty {
+    $$: XMLOrderedPropertyPart[],
+    [name: string]: XMLPropertyPart[]
+}
+
+interface XMLFunctionReturnPart {
+    $?: {
+        name?: string,
+    }
+}
+interface XMLOrderedFunctionReturnPart extends XMLFunctionReturnPart { "#name": string }
```

Comment:
Put a new line before this and spread it to multiple lines like the rest of the declarations.